apps/meteor/client/components/Omnichannel/utils/normalizeDepartments.ts

```diff
@@ -0,0 +1,20 @@
+import type { EndpointFunction } from '@rocket.chat/ui-contexts';
+
+import type { DepartmentListItem } from '../hooks/useDepartmentsList';
+
+export const normalizeDepartments = async (
+	departments: DepartmentListItem[],
+	selectedDepartment: string,
+	getDepartment: EndpointFunction<'GET', '/v1/livechat/department/:_id'>,
+): Promise<DepartmentListItem[]> => {
+	const isSelectedDepartmentAlreadyOnList = () => departments.some((department) => department._id === selectedDepartment);
+	if (!selectedDepartment || selectedDepartment === 'all' || isSelectedDepartmentAlreadyOnList()) {
+		return departments;
+	}
+
+	const { department: missingDepartment } = await getDepartment({});
+
+	return missingDepartment
+		? [...departments, { _id: missingDepartment._id, label: missingDepartment.name, value: missingDepartment._id }]
+		: departments;
+};
```

Comment:
Just to be on the safe side, might be a good idea to add a try catch for this fetch and guarantee that departments is returned even if something goes wrong.